test_runner/src/main/kotlin/ftl/reports/api/CreateTestExecutionData.kt

```diff
@@ -0,0 +1,39 @@
+package ftl.reports.api
+
+import com.google.api.services.testing.model.TestExecution
+import com.google.api.services.toolresults.model.ListTestCasesResponse
+import com.google.api.services.toolresults.model.Step
+import com.google.api.services.toolresults.model.Timestamp
+import ftl.gc.GcToolResults
+import ftl.reports.api.data.TestExecutionData
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.runBlocking
+
+internal fun List<TestExecution>.createTestExecutionDataListAsync(): List<TestExecutionData> = runBlocking {
+    map { testExecution ->
+        async(Dispatchers.IO) {
+            // calling this function in async block, speeds up execution because
+            // it is making api calls under the hood
+            testExecution.createTestExecutionData()
+        }
+    }.awaitAll()
+}
+
+private fun TestExecution.createTestExecutionData(): TestExecutionData {
+    val response: ListTestCasesResponse = GcToolResults.listTestCases(toolResultsStep)
+    val step: Step = GcToolResults.getStepResult(toolResultsStep)
```

Comment:
Both `GcToolResults.listTestCases` and `GcToolResults.getStepResult` are making api calls and seems that they don't depend on each other. Do you think we could make async call for each?

Reply:
Right, I will fix that.